src/Overlay.tsx

```diff
@@ -0,0 +1,90 @@
+import styled from 'styled-components'
+import React, {ReactElement} from 'react'
+import {get, COMMON, POSITION} from './constants'
+import {ComponentProps} from './utils/types'
+import { useOverlay, AnchoredPositionHookSettings, TouchOrMouseEvent } from './hooks'
+import Portal from './Portal'
+
+type StyledOverlayProps = {
+  width?: keyof typeof widthMap
+  height?: keyof typeof heightMap
+  visibility: 'visible' | 'hidden'
+}
+
+const heightMap = {
+  'sm': '480px',
+  'md': '640px',
+  'auto': 'auto'
+}
+
+const widthMap = {
+  'sm': '256px',
+  'md': '320px',
+  'lg': '480px',
+  'xl': '640px',
+  'auto': 'auto'
+}
+
+/*TODO replace with shadow functional color variable when it's shipped to primer/primitives*/
+const StyledOverlay  = styled.div<StyledOverlayProps>`
+  background-color: ${get('colors.bg.overlay')};
+  box-shadow: ${get('overlay.boxShadow')};
+  position: absolute;
+  min-width: 192px;
+  max-width: 480px;
+  height: ${props => heightMap[props.height || 'auto']};
+  width: ${props => widthMap[props.width || 'auto']};
+  border-radius: ${get('overlay.borderRadius')};
+  overflow: hidden;
+  animation: overlay-appear 200ms ${get('animation.easeOutCubic')};
+
+  @keyframes overlay-appear {
+    0% {
+      opacity: 0;
+      transform: translateY(${get('space.2')});
+    }
+    100% {
+      opacity: 1;
+    }
+  }
+  visibility: ${props => props.visibility};
+  ${COMMON};
+  ${POSITION};
+`
+
+export type OverlayProps = {
```

Comment:
Suggestion to add JSDoc-style comments to these props. At the very least it will help with IntelliSense, and if we ever want to auto-generate our documentation, these will be necessary. For now, I think it's fine to just copy what you've written in the documentation. In general anything that is exported should have JSDoc.

Reply:
done!

Reply:
we should add this to our Contributor Guidelines if we want to adopt this